### PR TITLE
Add DOM element locator

### LINKED ISSUE #225: Click on HTML element
Hello world!
I want to developp maybe a new feature for this wonderful project.
I was thinking that maybe 70% of the time we use this library form fill and submit form. 
I was thinking at a functionality to directly click on a html element in php.
Maybe something like `$page->click('.my-selector');`!
We can add this functionnality with a lot of function like :
- form submit `$page->submitForm('.my-selector');`
- fill input `$page->fillInput('.my-selector', 'some text');`*
- etc.

And maybe with a new object like that `$page->formHandler->fillInput('.my-selector');`

And maybe after do a whole html support with node, etc.

Thanks for reading this!
(and sorry for my bad english)

### PR DESCRIPTION
**Pull request by enricodias**
Allows searching elements using the new `find` method. Further work is needed to make it work on edge cases, such as if the element is outside the visible area to the left or top.

Resolves #225

### CODE REVIEW

**Comment by enricodias** [comment]
It's incredible how the StyleCI finishes running before my browser manages to load the page :)

**Comment by enricodias** [comment]
I'll have to rework those scroll methods in another PR. Having to increase the sleep time in the tests isn't a good solution.

**Comment by enricodias** [comment]
Finally, all tests passed.

Sorry for the mess with all those commits. It seems like these new tests with Symfony run slower than before. I couldn't reproduce that behavior locally and had to keep pushing here.

**Comment by GrahamCampbell** [comment]
Sounds good. Let's merge that other PR into 0.11, then I can merge upwards into 1.0, then this PR can be rebased (and kept targetting 1.0).

**Comment by AlexeyKosov** [comment]
Is there anything blocking this PR from merging? Looking forward to using the new functionality

**Comment by GrahamCampbell** [comment]
Are you able to rebase this @enricodias. I am going to do the 1.0.0 release now, so this PR will be for 1.1.0. :)

**Comment by enricodias** [comment]
@GrahamCampbell Done. I also removed the `usleep` calls since they are no longer needed.

**Comment by GrahamCampbell** [comment]
🎉

File: src/Input/Mouse.php [hunk 1]
```diff
@@ -207,6 +209,95 @@ private function scroll(int $distanceY, int $distanceX = 0): self
...
+
+        try {
+            $elementList = $this->page
+                ->evaluate('JSON.parse(JSON.stringify(document.querySelectorAll("'.$selectors.'")));')
+                ->getReturnValue();
+
+            $position = \max(0, (\count($elementList) - 1));
+
+            $element = $this->page
+                ->evaluate('JSON.parse(JSON.stringify(document.querySelectorAll("'.$selectors.'")['.$position.'].getBoundingClientRect()));')
```
Comment (momala454) [inline src/Input/Mouse.php:276]: shouldn't we escape the selector to escape " ?

Comment (enricodias) [reply src/Input/Mouse.php:276]: You mean to accept things like `->find('input[name="name"]');`?

Comment (momala454) [reply src/Input/Mouse.php:276]: yes

**Pull request merged**